validate.go

```diff
@@ -9,28 +9,51 @@ import (
 	"sync"
 	"time"
 
+	"github.com/fatih/color"
+
 	"github.com/aelsabbahy/goss/outputs"
 	"github.com/aelsabbahy/goss/resource"
 	"github.com/aelsabbahy/goss/system"
 	"github.com/aelsabbahy/goss/util"
-	"github.com/fatih/color"
-	"github.com/urfave/cli"
 )
 
-func getGossConfig(c *cli.Context) GossConfig {
+type RuntimeConfig struct {
```

Comment:
Arguably not the best way to go here, I'll see how it works out and refactor a bit while working on #544, for now it gets me moving with the other changes mentioned and already this is usable as a package - minus logging uglyness and os.Exit()

Reply:
If @aelsabbahy isn't bothered by this we can leave it here for now until we find a better home for it.

Reply:
This can probably move to a separate file.. but I'm fine with this being a later PR.